tests/src/test/spark330/scala/com/nvidia/spark/rapids/BloomFilterAggregateQuerySuite.scala

```diff
@@ -0,0 +1,102 @@
+  val funcId_bloom_filter_agg = new FunctionIdentifier("bloom_filter_agg")
+  val funcId_might_contain = new FunctionIdentifier("might_contain")
+
+  private def installSqlFuncs(spark: SparkSession): Unit = {
+    // Register 'bloom_filter_agg' to builtin.
+    spark.sessionState.functionRegistry.registerFunction(funcId_bloom_filter_agg,
+      new ExpressionInfo(classOf[BloomFilterAggregate].getName, "bloom_filter_agg"),
+      (children: Seq[Expression]) => children.size match {
+        case 1 => new BloomFilterAggregate(children.head)
+        case 2 => new BloomFilterAggregate(children.head, children(1))
+        case 3 => new BloomFilterAggregate(children.head, children(1), children(2))
+      })
+
+    // Register 'might_contain' to builtin.
+    spark.sessionState.functionRegistry.registerFunction(funcId_might_contain,
+      new ExpressionInfo(classOf[BloomFilterMightContain].getName, "might_contain"),
+      (children: Seq[Expression]) => BloomFilterMightContain(children.head, children(1)))
+  }
+
+  private def uninstallSqlFuncs(spark: SparkSession): Unit = {
+    spark.sessionState.functionRegistry.dropFunction(funcId_bloom_filter_agg)
+    spark.sessionState.functionRegistry.dropFunction(funcId_might_contain)
+  }
+
+  private def buildData(spark: SparkSession): DataFrame = {
+    import spark.implicits._
+    (Seq(Long.MinValue, 0, Long.MaxValue) ++ (1L to 10000L)).toDF("col")
+  }
+
+  for (numEstimated <- Seq(4096L, 4194304L, Long.MaxValue,
+    SQLConf.RUNTIME_BLOOM_FILTER_MAX_NUM_ITEMS.defaultValue.get)) {
+    for (numBits <- Seq(4096L, 4194304L, Long.MaxValue,
+      SQLConf.RUNTIME_BLOOM_FILTER_MAX_NUM_BITS.defaultValue.get)) {
+      ALLOW_NON_GPU_testSparkResultsAreEqual(
+        s"might_contain estimated=$numEstimated numBits=$numBits",
+        buildData,
+        Seq("ObjectHashAggregateExec", "ShuffleExchangeExec"))(df =>
+        {
+          val table = "bloom_filter_test"
+          val sqlString =
+            s"""
+               |SELECT might_contain(
+               |            (SELECT bloom_filter_agg(col,
+               |              cast($numEstimated as long),
+               |              cast($numBits as long))
+               |             FROM $table),
+               |            col) positive_membership_test,
+               |       might_contain(
+               |            (SELECT bloom_filter_agg(col,
+               |              cast($numEstimated as long),
+               |              cast($numBits as long))
+               |             FROM values (-1L), (100001L), (20000L) as t(col)),
+               |            col) negative_membership_test
+               |FROM $table
+              """.stripMargin
+          df.createOrReplaceTempView(table)
+          try {
+            installSqlFuncs(df.sparkSession)
+            df.sparkSession.sql(sqlString)
+          } finally {
+            uninstallSqlFuncs(df.sparkSession)
+          }
+        })
```

Comment:
I don't see any tests for nulls.  Null input values or null bloom filter. I know that these are corner cases that should not exist in our real world use cases, but I would like to make sure that we are covered because the CPU thinks that they need to support these cases.

Reply:
Null tests were added but are not passing currently because NVIDIA/spark-rapids-jni#1269 currently doesn't support null rows (actually throws an exception if they are seen).  It was built with the assumption input would always be from xxhash64 which cannot produce nulls.  However I agree it's prudent to handle nulls just in case this expression gets used outside of an xxhash64 context.  @nvdbaranec is currently updating the spark-rapids-jni  PR to handle nulls.